mev-relay-rs/src/relay.rs

```diff
@@ -62,22 +74,30 @@ fn validate_signed_block(
     signed_block: &mut SignedBlindedBeaconBlock,
     public_key: &BlsPublicKey,
     local_payload: &mut ExecutionPayload,
+    current_slot: Slot,
+    genesis_validators_root: Root,
+
     context: &Context,
 ) -> Result<(), Error> {
     let local_block_hash = local_payload.block_hash();
     let block_hash = signed_block.block_hash();
+    let slot = signed_block.slot();
     if block_hash != local_block_hash {
         return Err(Error::UnknownBlock)
     }
 
     // OPTIONAL:
     // -- verify w/ consensus?
     // verify slot is timely
-    // verify proposer_index is correct
+    if slot < current_slot || slot > current_slot + PROPOSAL_TOLERANCE_DELAY {
+        return Err(Error::UnknownBlock);
+    }
+
+    // verify proposer_index is correct 
+    // -> Isn't this checked by successfully returning the publickey?
```

Comment:
so for a given valid bid, it will sit on a particular fork of the chain, and it is possible for the proposer index to differ across the same slot in this case

so the intention here was to get our local view of the proposer and verify this bid was for the right validator